ui/src/app/components/add-index-form/add-index-form.component.spec.ts

```diff
@@ -1,36 +1,192 @@
 import { HttpClientModule } from '@angular/common/http';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { ReactiveFormsModule } from '@angular/forms';
-import { MatButtonModule } from '@angular/material/button';
-import { MatCardModule } from '@angular/material/card';
-import { MatDialogModule } from '@angular/material/dialog';
+import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
 import { MatFormFieldModule } from '@angular/material/form-field';
 import { MatInputModule } from '@angular/material/input';
 import { MatSelectModule } from '@angular/material/select';
 import { MatSnackBarModule } from '@angular/material/snack-bar';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
-
+import { of } from 'rxjs';
+import IRule from 'src/app/model/rule';
+import { ConversionService } from 'src/app/services/conversion/conversion.service';
+import { DataService } from 'src/app/services/data/data.service';
+import { SidenavService } from 'src/app/services/sidenav/sidenav.service';
+import mockIConv from 'src/mocks/conv';
 import { AddIndexFormComponent } from './add-index-form.component';
 
 describe('AddIndexFormComponent', () => {
   let component: AddIndexFormComponent;
   let fixture: ComponentFixture<AddIndexFormComponent>;
+  let dataServiceSpy: jasmine.SpyObj<DataService>;
+  let sidenavServiceSpy: jasmine.SpyObj<SidenavService>;
+  let fb: FormBuilder;
+  let conversionServiceSpy: jasmine.SpyObj<ConversionService>;
 
   beforeEach(async () => {
+    dataServiceSpy = jasmine.createSpyObj('DataService', ['conv', 'applyRule', 'dropRule']);
+    sidenavServiceSpy = jasmine.createSpyObj('SidenavService', [
+      'sidenavAddIndexTable',
+      'setSidenavAddIndexTable',
+      'closeSidenav',
+      'displayRuleFlag',
+      'ruleData',
+    ]);
+    conversionServiceSpy = jasmine.createSpyObj('ConversionService', [
+      'getTableIdFromSpName',
+      'getColIdFromSpannerColName',
+    ]);
+
     await TestBed.configureTestingModule({
-      declarations: [ AddIndexFormComponent ],
-      imports: [ReactiveFormsModule, HttpClientModule, MatSnackBarModule, MatSelectModule, BrowserAnimationsModule, MatFormFieldModule, MatInputModule]
+      declarations: [AddIndexFormComponent],
+      imports: [ReactiveFormsModule, HttpClientModule, MatSnackBarModule, MatSelectModule, BrowserAnimationsModule, MatFormFieldModule, MatInputModule],
+      providers: [
+        FormBuilder,
+        { provide: DataService, useValue: dataServiceSpy },
+        { provide: SidenavService, useValue: sidenavServiceSpy },
+        { provide: ConversionService, useValue: conversionServiceSpy },
+      ],
     })
-    .compileComponents();
+      .compileComponents();
+
+    fb = TestBed.inject(FormBuilder);
+
+    conversionServiceSpy.getColIdFromSpannerColName.and.returnValue("TestId")
+    conversionServiceSpy.getTableIdFromSpName.and.returnValue("t1")
```

Comment:
The method signature is `getTableIdFromSpName(name: string, conv: IConv)`.

Does this mean whatever the input parameters, it will always return `t1`?